app/domain/README.md

```diff
@@ -0,0 +1,296 @@
+# Why `app/domain`?
+
+## Introduction
+
+This document explains the coding standards used in `app/domain`, and the
+motivation behind them.  By design, these standards are differerent from those
+of a typical Rails app. 
+
+You can think of `app/domain` as a haven where pure Conjur business logic (the
+Conjur "domain") lives as plain ruby code, unpolluted by framework-specific
+concepts like controllers or routes.  The code in this folder should have no
+dependencies on Rails itself (if you find any, they are likely mistakes we
+should fix).  
+
+Currently, `app/domain` contains mostly authentication logic.  But its scope is
+not limited to authenticators.
+
+At the highest level, this approach aims for:
+
+- Explicit, maintainable code that is easy to understand
+- Fast unit tests
+- Intuitive names that correspond with our documentation and the way we
+  naturally talk
+
+We'll dig into the technical specifics below.
+
+## The "Functional Object-Oriented" Style
+
+I'll offer a TLDR version here.  For more in depth discussion of the same
+principles, the video [Functional Architecture for the Practical
+Rubyist](https://www.youtube.com/watch?v=7qnsRejCyEQ&feature=youtu.be) is a
+good place to start.
+
+The basic idea is to reap many benefits of functional programming while 
+remaining true to ruby's objected-oriented paradigm.
+
+In brief, this means building features out of three basic kinds of objects, and
+maintaing a strict sepration of responsibilities:
+
+- **Command objects**
+  - These go by many names: service objects, interactors, use cases
+  - Coordinates high-level logic involving multiple collaborators
+    - For example: `CreateUser` might need to coordinate user input validation,
+      a database transaction, a log entry, and a confirmation email
```

Comment:
i would add here that `CreateUser` will consume some of its logic from other Command objects with smaller resplonsibility